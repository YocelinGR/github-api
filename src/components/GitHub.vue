<template>
<div>
  <h1>GitHub</h1>
  <div>
      <button @click="mutationHeart()">click to inserte EYES</button>
      <p>Creaci√≥n:  </p>
      <span>{{ repository.id }}</span>
  </div>
</div>
</template>

<script>
import gql from 'graphql-tag';
 const mutationEmoji = gql`
 mutation AddReactionToIssue($input: AddReactionInput!) {
   addReaction(input: $input) {
     reaction {
       content
     }
     subject {
       id
     }
   }
 }
 `;
const repository = gql`
  query FindIssueID($owner: String!, $name: String!) {
  repository(owner: $owner, name:$name) {
    id
  }
 }
`;

export default {
  data: () => ({
    repository: {}
  }),
    methods: {
      mutationHeart(){
        const mutationParams = {
            subjectId:"MDU6SXNzdWUzMzYyNDY5NDA=",
            content: 'EYES',
            clientMutationId: "MDQ6VXNlcjM5ODMzMDQ1",
        };
        location.reload();
        this.$apollo.mutate({
          mutation: mutationEmoji,
          variables: {
            input: mutationParams,
          }
        })
      }
    },
  apollo: {
    repository: {
      query: repository,
      variables() {
        return{
          owner: 'YocelinGR',
          name: 'cdmx-2018-06-bc-core-am-data-dashboard',
        }
      }
    }
  }
}
</script>
