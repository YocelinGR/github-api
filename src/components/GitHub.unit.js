import GitHub from './GitHub.vue'
import { createLocalVue } from '@vue/test-utils'

describe('@components/GitHub', () => {
  const localVue = createLocalVue()
  it('GitHub is a vue instance', () => {
    const wrapper = shallowMount(GitHub, { localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('called Apollo mutations', () => {
    const mutate = jest.fn()
    const wrapper = mount(GitHub, {
      localVue,
      mocks: {
        $apollo: {
          mutate,
        },
      },
    })
    wrapper.vm.mutationHeart()
    expect(mutate).toBeCalled()
  })

  it('Displays user starred repositories data', () => {
    const wrapper = shallowMount(GitHub, { localVue })
    wrapper.setData({
      repository: {
        id: "MDEwOlJlcG9zaXRvcnkxMzc5MjE1NjY="
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
