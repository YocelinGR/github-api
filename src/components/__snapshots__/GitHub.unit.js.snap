// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/GitHub Displays user starred repositories data 1`] = `
<div>
  <h1>
    GitHub
  </h1>
   
  <div>
    <button>
      click to inserte EYES
    </button>
     
    <p>
      Creaci√≥n:  
    </p>
     
    <span>
      MDEwOlJlcG9zaXRvcnkxMzc5MjE1NjY=
    </span>
  </div>
</div>
`;
